src/main/io/osd.c

```diff
@@ -55,522 +60,290 @@
-    {"DTERM LPF", OME_UINT16, NULL, &entryDtermLpf},
-    {"YAW LPF", OME_UINT16, NULL, &entryYawLpf},
-    {"YAW P LIMIT", OME_UINT16, NULL, &entryYawPLimit},
-    {"MINTHROTTLE", OME_UINT16, NULL, &entryMinThrottle},
-    {"VBAT SCALE", OME_UINT8, NULL, &entryVbatScale},
-    {"VBAT CELL MAX", OME_UINT8, NULL, &entryVbatMaxCell},
-    {"BACK", OME_Back, NULL, NULL},
-    {NULL, OME_END, NULL, NULL}
-};
+            if (FLIGHT_MODE(FAILSAFE_MODE))
+                p = "!FS";
+            else if (FLIGHT_MODE(ANGLE_MODE))
+                p = "STAB";
+            else if (FLIGHT_MODE(HORIZON_MODE))
+                p = "HOR";
 
-OSD_UINT8_t entryPidProfile = {&masterConfig.current_profile_index, 0, MAX_PROFILE_COUNT, 1};
+            max7456Write(elemPosX, elemPosY, p);
+            return;
+        }
 
-OSD_Entry menuImu[] =
-{
-    {"-----CFG. IMU-----", OME_Label, NULL, NULL},
-    {"PID", OME_Submenu, osdChangeScreen, &menuPid[0]},
-    {"PID PROFILE", OME_UINT8, NULL, &entryPidProfile},
-    {"RATE & RXPO", OME_Submenu, osdChangeScreen, &menuRateExpo[0]},
-    {"RC PREVIEW", OME_Submenu, osdChangeScreen, &menuRc[0]},
-    {"MISC", OME_Submenu, osdChangeScreen, &menuMisc[0]},
-    {"BACK", OME_Back, NULL, NULL},
-    {NULL, OME_END, NULL, NULL}
-};
+        case OSD_CRAFT_NAME:
+        {
+            if (strlen(masterConfig.name) == 0)
+                strcpy(buff, "CRAFT_NAME");
+            else {
+                for (uint8_t i = 0; i < MAX_NAME_LENGTH; i++) {
+                    buff[i] = toupper((unsigned char)masterConfig.name[i]);
+                    if (masterConfig.name[i] == 0)
+                        break;
+                }
+            }
 
-uint8_t tempPid[4][3];
+            break;
+        }
 
-static OSD_UINT8_t entryRollP = {&tempPid[PIDROLL][0], 10, 150, 1};
-static OSD_UINT8_t entryRollI = {&tempPid[PIDROLL][1], 1, 150, 1};
-static OSD_UINT8_t entryRollD = {&tempPid[PIDROLL][2], 0, 150, 1};
+        case OSD_THROTTLE_POS:
+        {
+            buff[0] = SYM_THR;
+            buff[1] = SYM_THR1;
+            sprintf(buff + 2, "%d", (constrain(rcData[THROTTLE], PWM_RANGE_MIN, PWM_RANGE_MAX) - PWM_RANGE_MIN) * 100 / (PWM_RANGE_MAX - PWM_RANGE_MIN));
+            break;
+        }
 
-static OSD_UINT8_t entryPitchP = {&tempPid[PIDPITCH][0], 10, 150, 1};
-static OSD_UINT8_t entryPitchI = {&tempPid[PIDPITCH][1], 1, 150, 1};
-static OSD_UINT8_t entryPitchD = {&tempPid[PIDPITCH][2], 0, 150, 1};
+#ifdef VTX
+        case OSD_VTX_CHANNEL:
+        {
+            sprintf(buff, "CH:%d", current_vtx_channel % CHANNELS_PER_BAND + 1);
+            break;
+        }
+#endif // VTX
 
-static OSD_UINT8_t entryYawP = {&tempPid[PIDYAW][0], 10, 150, 1};
-static OSD_UINT8_t entryYawI = {&tempPid[PIDYAW][1], 1, 150, 1};
-static OSD_UINT8_t entryYawD = {&tempPid[PIDYAW][2], 0, 150, 1};
+        case OSD_CROSSHAIRS:
+        {
+            uint8_t *screenBuffer = max7456GetScreenBuffer();
```

Comment:
With max7456WriteChar, it would look like:
```
        case OSD_CROSSHAIRS:
            elemPosX = 14;
            elemPosY = 6;
            if (maxScreenSize == VIDEO_BUFFER_CHARS_PAL)
                ++elemPosY;
            max7456WriteChar(elemPosX - 1, elemPosY, SYM_AH_CENTER_LINE);
            max7456WriteChar(elemPosX    , elemPosY, SYM_AH_CENTER);
            max7456WriteChar(elemPosX + 1, elemPosY, SYM_AH_CENTER_LINE_RIGHT);
            return;
```
Is this what you mean?